docs/crio.conf.5.md

```diff
@@ -138,6 +138,31 @@ the container runtime configuration.
   Changes the meaning of an empty seccomp profile.  By default (and according to CRI spec), an empty profile means unconfined.
   This option tells CRI-O to treat an empty profile as the default profile, which might increase security.
 
+**seccomp_use_notifier**=false
+  Enable the seccomp notifier feature.
+
+  This feature can help you to debug seccomp related issues, for example if
+  blocked syscalls (permission denied errors) have negative impact on the
+  workload.
+
+  To be able to use this feature, just enable it via the flag below and
+  configure a runtime which has the annotation
+  "io.kubernetes.cri-o.seccompNotifier" in the `allowed_annotations` array.
+
+  It also requires at least runc 1.1.0 or crun 0.19 which support the notifier
+  feature.
+
+  If everything is setup, CRI-O will modify chosen seccomp profiles for
+  containers if the annotation "io.kubernetes.cri-o.seccompNotifier" is set on
+  the Pod sandbox. CRI-O will then get notified if a container is using a
```

Comment:
what is the difference with cri-o terminating because of a notification vs a pod failing on EPERM from a blocked syscall? I wonder if we should notify in a different way (metric and cri-o log, maybe?)

Reply:
We cannot find the root cause of the pod termination easily because audit does not logs act kill for example.

A metric is interesting, but we will only catch the first wrong syscall because then the termination happens.

I was thinking if it would make sense to collect the wrong syscalls (until when?) and then throw the metrics. Hm… :thinking:

Reply:
from my perspective, the notify feature should be used to do information collection. I'm thinking it'd be used for syscalls that folks aren't worried about, but they want to tighten the syscalls allowed to reduce attack surface. Maybe as the second step, after using the bpf hook for a bit to filter the commonly used ones. The remaining (non dangerous) could be set to notify, and cri-o would notify if they're used. If a notification happens, the user could update the profile to block or allow, depending on how they feel about the syscall.

Reply:
Alright, I see the following use cases for now:

- Removing unnecessary syscalls from my already existing seccomp profile
- Validating if a profile has negative impact on my application

Both could be achieved via metrics and logging, but would require manual monitoring of them. The container stop is immediately visible to users, which I see as a benefit. How about:

- Adding metrics for the notifications
- Making the container stop behavior optional

Reply:
Made the container stop behavior configurable. I think we should not add metrics since all of them would have a high cardinality.